lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -0,0 +1,323 @@
+  return new_p;
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_context_get()                                             **
+ **                                                                        **
+ ** Description: Get the amf context based on ue identity                  **
+ **                                                                        **
+ ** Input : ue_id : user equipment identity value                          **
+ **                                                                        **
+ ** Return: amf_context structure,Success case                             **
+ **         NULL ,Failure case                                             **
+ ***************************************************************************/
+amf_context_t* amf_context_get(const amf_ue_ngap_id_t ue_id) {
+  amf_context_t* amf_context_p = nullptr;
+
+  if (INVALID_AMF_UE_NGAP_ID != ue_id) {
+    ue_m5gmm_context_s* ue_mm_context =
+        amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+
+    if (ue_mm_context) {
+      amf_context_p = &ue_mm_context->amf_context;
+    }
+    OAILOG_DEBUG(LOG_NAS_AMF, "Stored UE id " AMF_UE_NGAP_ID_FMT " \n", ue_id);
+  }
+  return amf_context_p;
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_ue_context_exists_amf_ue_ngap_id()                        **
+ **                                                                        **
+ ** Description: Checks if UE context exists already or not                **
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+ue_m5gmm_context_s* amf_ue_context_exists_amf_ue_ngap_id(
+    const amf_ue_ngap_id_t amf_ue_ngap_id) {
+  std::unordered_map<amf_ue_ngap_id_t, ue_m5gmm_context_s*>::iterator
+      found_ue_id = ue_context_map.find(amf_ue_ngap_id);
+
+  if (found_ue_id == ue_context_map.end()) {
+    return NULL;
+  } else {
+    return found_ue_id->second;
+  }
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_insert_smf_context()                                      **
+ **                                                                        **
+ ** Description: Insert smf context in the vector                          **
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+smf_context_t* amf_insert_smf_context(

```

Comment:
This whole function definition looks very convoluted and the description is not really accurate. From what I see, a fresh new SMF context is created inside the function, a matching pdu session context is replaced by it and a pointer to the element is returned. Why? This function effectively overwrites an existing entry and returns a pointer to the session context.